### PR TITLE
Copy button added to details modal

### PR DESCRIPTION
**Pull request by nakedmcse**
Add a copy button on useful details modals field.  When icon is clicked it copies text to clipboard.  Icon resets after two seconds.

### CODE REVIEW

File: src/model/db/FileUpload.model.ts [hunk 1]
```diff
@@ -96,6 +96,16 @@ export class FileUploadModel extends AbstractModel {
         return FileUtils.isFileExpired(this);
     }
 
+    public get fileProtectionLevel(): string {
+        if (this.encrypted) {
+            return "Encrypted";
+        }
+        if (this.settings?.password != null) {
```
Comment (VictoriqueMoe) [inline src/model/db/FileUpload.model.ts:103]: can just be ` if (this.settings?.password) {`

**Review by VictoriqueMoe — CHANGES_REQUESTED** [review]

File: src/model/db/FileUpload.model.ts (hunk 1 above)
Comment (nakedmcse) [reply src/model/db/FileUpload.model.ts]: Implemented

File: src/model/rest/FileEntry.ts [hunk 2]
```diff
@@ -41,6 +41,9 @@ export class FileEntry {
     @Nullable(String)
     public mediaType: string | null = null;
 
+    @Property()
+    public fileProtectionLevel: string;
```
Comment (VictoriqueMoe) [inline src/model/rest/FileEntry.ts:45]: I think this should be a union string 

```ts
type prtectionlevel = "Encrypted" | "Password" | "None";
```
where you put the type, you decide what is best.

File: src/model/db/FileUpload.model.ts [hunk 3]
```diff
@@ -96,6 +96,16 @@ export class FileUploadModel extends AbstractModel {
         return FileUtils.isFileExpired(this);
     }
 
+    public get fileProtectionLevel(): string {
```
Comment (VictoriqueMoe) [inline src/model/db/FileUpload.model.ts:99]: this method should return the union you will create

**Review by VictoriqueMoe — CHANGES_REQUESTED** [review]

File: src/model/rest/FileEntry.ts (hunk 2 above)
Comment (nakedmcse) [reply src/model/rest/FileEntry.ts:45]: Added to typeings.

Comment (nakedmcse) [reply src/model/rest/FileEntry.ts:45]: Adjusted

**Review by VictoriqueMoe — APPROVED** [review]

**Pull request merged**